tracpro/msgs/views.py

```diff
@@ -112,7 +112,21 @@ class List(OrgPermsMixin, SmartListView):
         def derive_queryset(self, **kwargs):
             qs = InboxMessage.get_all(self.request.org, self.request.data_regions)
             qs = qs.select_related('contact')
-            qs = qs.order_by('contact', '-created_on').distinct('contact')
+
+            qs = qs.order_by('contact', '-created_on')
+            qs = qs.distinct('contact')
+
+            return qs
+
+        def order_queryset(self, qs):
+            """
+            Override the order_queryset() method from smartmin
```

Comment:
I don't think this comment is very helpful - maybe something more specific about how the default ordering mechanisms won't work due to the distinct query on contact?

Should we also consider applying validation of the `_order` parameter?